---
title: "**Choosing Genomics Tools: Gene Pattern Activity**"
output:
  html_document:
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 3
    number_sections: false
css: style.css
editor_options:
  markdown:
    wrap: 72
---

## Bulk RNA-seq differential expression analysis

The datasets for this activity consist of 20 Breast cancer primary tumor samples with 20 normal matched samples from TCGA as explained in the study ["Molecular analysis of TCGA breast cancer histologic types"](https://pmc.ncbi.nlm.nih.gov/articles/PMC9028992/).

The aim of this activity is to answer the question -- "What genes differentiate samples with breast cancer from samples without breast cancer?". This is an example of differential expression analysis, a downstream analysis step in the bulk RNA-seq workflow.

### Download Data and log into the Gene Pattern platform

<input type="checkbox"> [Download the relevant datasets](https://datasets.genepattern.org/?prefix=data/workshops/240411-PSTP/)<br/>
  - <input type="checkbox"> Download `BRCA_HUGO_symbols.preprocessed.cls`<br/>
  - <input type="checkbox"> Download `BRCA_HUGO_symbols.preprocessed.gct`<br/>

```{r echo=FALSE, fig.alt="Download the c l s and g c t files from the gene pattern datasets resource, specifically the 2 4 0 4 1 1 hyphen P S T P workshops directory", fig.link="https://datasets.genepattern.org/?prefix=data/workshops/240411-PSTP/"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM/edit#slide=id.g2f2ca03ec26_0_0")
```

*Make sure you know where they saved on your computer as you'll be uploading them in an upcoming step.*

<details><summary>What are these file types?</summary>

The `gct` file is a Gene Cluster Text file format used to describe an expression dataset. [See more at this Broad Institute resource](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GCT:_Gene_Cluster_Text_file_format_.28.2A.gct.29)

```{r echo=FALSE, fig.alt="Description of g c t file from the Broad Institute", fig.link="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GCT:_Gene_Cluster_Text_file_format_.28.2A.gct.29"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM/edit#slide=id.g2f2ca03ec26_0_11")
```

The `cls` file is a Categorial (e.g., tumor vs normal) class file format, specifying phenotype or class labels associated with each sample in the expression data. [See more at this Broad Institute resource](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29)

```{r echo=FALSE, fig.alt="Description of c l s file from the Broad Institute", fig.link="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM/edit#slide=id.g2f2ca03ec26_0_19")
```

[See this resource from GenePattern for more information on creating these input files for your work](https://www.genepattern.org/file-formats-guide/#gsc.tab=0)

```{r echo=FALSE, fig.alt="Description of c l s and g c t file types with a link to a resource on creating these file types for your work", fig.link="https://www.genepattern.org/file-formats-guide/#gsc.tab=0"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM/edit#slide=id.g2f2ca03ec26_0_5")
```

</details>

<input type="checkbox"> [Go to the Gene Pattern platform](http://cloud.genepattern.org/gp/) and sign in

### Run the ComparativeMarkerSelection Module

<input type="checkbox"> Find the "ComparativeMarkerSelection" Module

```{r echo=FALSE, fig.alt="Find the Comparative Marker Selection Module to identify differentially expressed genes that can discriminate between our two groups of data"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM/edit#slide=id.g2f2ca03ec26_0_26")
```

<input type="checkbox"> Select the `gct` file (`BRCA_HUGO_symbols.preprocessed.gct`) for the "Input file*" option.

<input type="checkbox"> Select the `cls` file (`BRCA_HUGO_symbols.preprocessed.cls`) for the "cls file*" option.

```{r echo=FALSE, fig.alt="select the input files"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM/edit#slide=id.g2f2ca03ec26_0_41")
```


<!--
## Single cell RNA-seq

- [Download practice 10x single cell data](https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz)
- [Go to the Gene Pattern platform](http://cloud.genepattern.org/gp/)
-->
