---
title: "Bonus activity: Using the command-line for single-cell Spatial Transcriptomics analysis"
output:
  html_document:
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 3
    number_sections: false
css: style.css
editor_options:
  markdown:
    wrap: 72
---

```{r echo = FALSE, show = FALSE}
knitr::opts_chunk$set(number_sections = FALSE)
```

To continue to practice using `spatialGE` using R, you can access our PositCloud instance until April 27th. See the [clinical workshop activity guidance](https://hutchdatascience.org/UMich_ITN_Workshop/clinical_data_activity.html) for instructions on how to connect.

Alternatively you can download and install R and RStudio to use on your local machine or see if your institution has a sever available with R and R Studio.

See this [link for information on installations](https://jhudatascience.org/intro_to_r/modules/Setup/Setup.html). 


The package ***`spatialGE`*** R package and [web application](https://spatialge.moffitt.org)
can be used to detect spatial patterns in gene expression at the gene and gene set level.
Data from multiple spatial transcriptomics platforms can be analyzed, as long as
gene expression counts per cell are associated with spatial coordinates of
those cells. Both ***`spatialGE`*** R package and web application are also
compatible with outputs from the CosMx-SMI spatial single-cell platform.

In this tutorial, the functions `STenrich`, `STclust`, `STdiff`, and `STgradient`
will be used  to test for spatial gene set enrichment and genes showing expression
gradients in a [CosMx-SMI data set](https://nanostring.com/products/cosmx-spatial-molecular-imager/nsclc-ffpe-dataset/)
from Non-Small Cell Lung Cancer (NSCLC).

####  {.click_to_expand_block}

<details>

<summary>How is `spatialGE` installed?</summary>

The `spatialGE` repository is available at GitHub and can be installed via `devtools`.
To install `devtools` (in case it is not already installed in your R), please run
the following code:

```{r install_devtools, eval=FALSE}
if("devtools" %in% rownames(installed.packages()) == FALSE){
  install.packages("devtools")
}
```

After making sure `devtools` is installed, proceed to install `spatialGE`:

```{r install_spatialGE, eval=FALSE}
devtools::install_github("fridleylab/spatialGE")
```

</details>

####

## Get data files

The CosMx-SMI platform generates single-cell level gene expression with associated
x and y locations of the cells where measurements were taken. For the data set used in
this tutorial, RNA counts were measured for 960 genes in about 800K cells. The data
was generated from eight tissue slices, with counts obtained for a series of Fields of
Vision (FOVs) within each slice. However, in this tutorial, the data set has been
subset to 6 FOVs originating from two of the slices.

The complete data set can be downloaded from the links below. It is also likely
that registration in the website is necessary in order to download the data set.

-   [Lung 5-1](https://nanostring.com/resources/smi-ffpe-dataset-lung5-rep1-data/)
-   [Lung 6](https://nanostring.com/resources/smi-ffpe-dataset-lung6-data/)

**The data from the 6 FOVs used in this tutorial** is available at the [spatialGE_Data GitHub repository](https://github.com/FridleyLab/spatialGE_Data):

<input type="checkbox"> Download the files for this activity clicking
here:
<https://github.com/FridleyLab/spatialGE_Data/raw/refs/heads/main/nanostring_nsclc.zip?download=>\
<input type="checkbox"> Put this file on your desktop so it is easily
findable.\
<input type="checkbox"> Double click the zip file (or right click and
choose "unzip" or "decompress" to unzip the file.\
<input type="checkbox"> Open up your activity files you downloaded so we
can see what's there.

### Get familiar with the data set

Within the folder, we should a see two sub-folders, one for each of the two tissue
slices. The files within each sub-folder are:

-   Lung5_Rep1-Flat_files_and_images
    -   21_00732_LI_SING_filtered_feature_bc_matrix.h5 (the gene
        expression data)
    -   spatial
        -   CellComposite (tissue images)
        -   Lung5_Rep1_exprMat_file_SUBSET.csv (gene expression data)
        -   Lung5_Rep1_metadata_file_SUBSET.csv (cell x,y coordinates)
-   Lung6-Flat_files_and_images
    -   ...

Then load the spatialGE package:

```{r eval=FALSE}
library('spatialGE')
```

Now, specify the directory paths to the folders containing the data, assuming the
files are in the computer's desktop:

```{r eval=FALSE}
counts_fp = c('~/Desktop/nanostring_nsclc/Lung5_Rep1-Flat_files_and_images/Lung5_Rep1_exprMat_file_SUBSET.csv',
              '~/Desktop/nanostring_nsclc/Lung6-Flat_files_and_images/Lung6_exprMat_file_SUBSET.csv')

coordinates_fp = c('~/Desktop//nanostring_nsclc/Lung5_Rep1-Flat_files_and_images/Lung5_Rep1_metadata_file_SUBSET.csv',
                   '~/Desktop/nanostring_nsclc/Lung6-Flat_files_and_images/Lung6_metadata_file_SUBSET.csv')
```

Finally, create an STlist. The STlist is the R object in spatialGE that holds the
data and results.

**Important:** Note that the argument `samples` contains the names of each tissue
slice. These names must match partially the file paths passed to `rnacounts` and
`spotcoords`.

```{r eval=FALSE}
lc_stlist = spatialGE::STlist(rnacounts=counts_fp, spotcoords=coordinates_fp,
                              samples=c('Lung5_Rep1', 'Lung6'))
```

####  {.click_to_expand_block}

<details>

<summary>Uploading a CosMx experiment to the spatialGE web application</summary>

**Start a new project**

<input type="checkbox"> Click the blue "New Project" button.\
<input type="checkbox"> For
`What spatial transcriptomics platform are you using for this project?`
choose `CosMx-SMI`.\
<input type="checkbox"> Make your own project name and description
that's sensible. Could be something related to the workshop such as
"ITN-UMichigan_cosmx".\
<input type="checkbox"> Then click "Create".

**Uploading data**

<input type="checkbox"> For `Sample Name` put the ID indicating on the
folder, e.g. `Lung5_Rep1`.\
<input type="checkbox"> For the `Gene expression` box upload the file
`Lung5_Rep1_exprMat_file_SUBSET`. You can upload files by dragging and dropping or
by clicking on them to navigate.\
<input type="checkbox"> For the `Coordinates` box upload the file
`Lung5_Rep1_metadata_file_SUBSET.csv`.

<input type="checkbox"> Once the above steps are done click the green
`Import Sample`.

Now repeat these steps for the other tissue slice.

*You can use this checklist to keep track as you upload and follow the
steps for each sample.*

<input type="checkbox"> `Lung5_Rep1` data entered\
<input type="checkbox"> `Lung6` data entered

**NOTE**: Make sure to upload all the tissue slices before clicking the
`Import Data` button. You will not be able to edit the project (unless
you start a new project completely) after you click `Import Data`.

<input type="checkbox"> Make sure everything is as you intend and then
click `Import Data`.

This may take a little bit of time. Note you can have it send you an
email instead of waiting on the page.

</details>

####

Once a STlist object has been created, one can obtain count statistics using the
`summarize_STlist` function:

```{r count_stats, eval=FALSE}
summarize_STlist(lc_stlist)
```

Some cells have zero counts. We can look at the distribution of counts per cell for the first five and last five FOVs using the `distribution_plots` function:

```{r count_dstr, eval=FALSE}
cp <- distribution_plots(lc_stlist, plot_type='violin', plot_meta='total_counts')
cp[['total_counts']]
```

Cells with zero counts, can be removed using the `filter_data` function. The function
can also be used to remove all counts from specific genes. This option is useful in
this case, as CosMx-SMI panels include negative probe genes. While the negative
probes can be used in normalization of counts, in this tutorial those genes will
be removed. Negative probes in CosMx-SMI begin with the token "NegPrb":

```{r filter_chunk, eval=FALSE}
lc_stlist <- filter_data(lc_stlist, spot_minreads=20, rm_genes_expr='^NegPrb')
```

## Transformation of spatially-resolved transcriptomics data

The function `transform_data` allows data transformation using one of two possible
options. The first options applies log-transformation to the counts, after library
size normalization performed on each sample separately. The second option applies
variance-stabilizing transformation (SCT; @hafemeister_2019), which is a method
increasingly used in single-cell and spatial transcriptomics studies. To apply SCT
transformation, use the following command:

```{r norm_chunk, message=F, eval=FALSE}
lc_stlist <- transform_data(lc_stlist, method='sct')
```

## Detecting gene sets with spatial aggregation patterns

An important part of gene expression analysis is the use of gene sets to make
inferences about the functional significance of changes in expression. Normally,
this is achieved by conducting a Gene Set Enrichment Analysis (GSEA). While GSEA
can be completed in a similar fashion to scRNA-seq, it is possible with the `STenrich`
function to test for gene sets that show spatial non-uniform enrichment. In other
words, `STenrich` tests whether expression of a gene set is concentrated in one or
few areas of the tissue.

The `STenrich` function first calculates Euclidean distances among all cells.
Then, the gene expression of the sample is subset to the genes within the gene set
being tested. If too few genes are left after subset (`min_genes`), then the gene
set is omitted for that sample. The enrichment scores and standard deviations of
those genes are calculated for each cell. Next, cells with gene set scores
above the average gene set score across all cells are identified. The threshold to
define these high gene set score cells is defined by the average
gene set score plus a number (`min_sds`) of standard deviations. The sum of the
Euclidean distances between the high score cells is calculated.

The next step involves a permutation process, in which a null distribution is
generated in order to test if the (sum of) distances among high expression cells
are smaller than expected. To that end, a random sample of cells (regardless of
expression) is selected. The random sample has the same size as the number of high
expression cells. Then, the sum of distances among the randomly sampled cells is
calculated. The random selection is repeated as many times as requested (`reps``).
Finally, a p-value is calculated by noting how many times the sum of random distances
was higher than the sum of distances among high expression cells. If the sum of
random distances was most of the times higher than the sum of distances among high
expression cells, then the null hypothesis of no spatial aggregation is rejected
(i.e., cells with high gene set expression are more aggregated than expected by chance).

The first step to run `STenrich` is to obtain a list of gene sets. Here, the `msigdbr`
package will be used to obtain HALLMARK gene sets:

```{r genesets, message=F, eval=FALSE}
# Load msigdbr
library('msigdbr')

# Get HALLMARK gene sets
gene_sets <- msigdbr(species='Homo sapiens')
# Select the appropriate column depending on the msigdbr version
gs_column <- grep('gs_cat|gs_collection', colnames(gene_sets), value=T)[1]
gene_sets <- gene_sets[gene_sets[[gs_column]] == "H", ]

# Convert gene set dataframe to list
# The result is a named list. The names of the list are the names of each gene set
# The contents of each list element are the gene names within each gene set
gene_sets <- split(x=gene_sets[['gene_symbol']], f=gene_sets[['gs_name']])
```

For the purpose of this example, we will only test "signaling" pathways:
```{r genesets_sub, message=F, eval=FALSE}
gene_sets = gene_sets[grep('SIGNALING', names(gene_sets))]
```

Please keep in mind that it is assumed that HUGO gene names are used. If spatial
transcriptomics data from a non-human species is used, the appropriate database
should be used. Otherwise, `spatialGE` will not be able to identify genes within
the gene set.

The `STenrich` function can now be used. For this analysis, 1000 replicates (`reps=1000`)
will be used to generate the null distribution. In addition, for a gene set to be
considered as highly expressed in cell, it should be expressed at least over the
average gene set score plus 1.5 standard deviations (`num_sds=1.5`). Gene sets with
less than five genes in the sample (`min_genes=5`) and gene sets with less than 10
highly expressed cells (`min_units=10`) will not be tested. Notice a seed was set
(`seed=12345`), which critical for replication of the results in the future.

```{r stenrich, eval=FALSE}
stenrich_df <- STenrich(lc_stlist, gene_sets=gene_sets,
                        reps=1000,
                        num_sds=1.5,
                        min_genes=5,
                        min_units=10,
                        seed=12345)
```

The result of the function is a list of data frames, with one data frame per
samples (FOV in this case). For example, this is the data frame for the first
FOV in the STlist:

```{r stenrich_df, eval=FALSE}
head(stenrich_df[['Lung5_Rep1_fov_2']])
```

Each row represents a test for the null hypothesis of no spatial aggregation in
the expression of the set in the "gene_set" column. The column "size_test" is the
number of genes of a gene set that were present in the FOV. The larger this number
the better, as it indicates a better representation of the gene set in the sample.
The "adj_p_value" is the multiple test adjusted p-value, which is the value used to
decide if a gene set shows significant indications of a spatial pattern (adj_p_value < 0.05).

With a few lines of code, a visual summary can be generated that presents the gene
sets with adj_p_value < 0.05 across FOVs for each tissue slide:

```{r message=F, eval=FALSE}
# Load tidyverse for data frame manipulation
library('tidyverse')

# Combine all samples in a single data frame
# Subset to gene sets showing significant evidence of spatial pattern (adj_p_value < 0.05) and
# proportion higher or equal than 0.3 of genes from a gene set present in sample
res <- bind_rows(stenrich_df) %>%
  mutate(prop_gene_set=size_test/size_gene_set) %>%
  filter(prop_gene_set >= 0.3 & adj_p_value < 0.05) %>%
  mutate(slide=str_extract(sample_name, "Lung5_Rep1|Lung6")) %>%
  select(slide, gene_set) %>%
  mutate(gene_set=str_replace(gene_set, 'HALLMARK_', ''))

# Generate barplot showing the number of FOVs with siginificant evidence of spatial
# patterns for each tissue slide
ggplot(res) +
  geom_bar(aes(x=gene_set)) +
  xlab(NULL) +
  theme(axis.text.x=element_text(angle=70, vjust=1, hjust=1)) +
  facet_wrap(~slide)
```

The "NOTCH signaling" pathway showed spatial patterns less frequently in the slide
"Lung5_Rep1" compared to "Lung6". With the function `STplot`, the average gene set
expression (not GSEA score) for this pathway can be plotted. First, FOVs with and
without NOTCH  signaling spatial patterns will be identified from "Lung5_Rep1" and
"Lung6" respectively.

```{r notch_sig1, eval=FALSE}
bind_rows(stenrich_df) %>%
  mutate(prop_gene_set=size_test/size_gene_set) %>%
  filter(prop_gene_set >= 0.3 & adj_p_value >= 0.05) %>%
  arrange(desc(prop_gene_set), desc(adj_p_value)) %>%
  filter(gene_set == 'HALLMARK_NOTCH_SIGNALING') %>%
  filter(str_detect(sample_name, "Lung5_Rep1")) %>%
  slice_head(n=3)
```

```{r notch_sig2, eval=FALSE}
bind_rows(stenrich_df) %>%
  mutate(prop_gene_set=size_test/size_gene_set) %>%
  filter(prop_gene_set >= 0.3 & adj_p_value < 0.05) %>%
  arrange(desc(prop_gene_set), adj_p_value) %>%
  filter(gene_set == 'HALLMARK_NOTCH_SIGNALING') %>%
  filter(str_detect(sample_name, "Lung6")) %>%
  slice_head(n=3)
```

Plots can be generated like so:

```{r notch_plots, fig.width=10, eval=FALSE}
qp <- STplot(lc_stlist, genes=gene_sets[names(gene_sets) == 'HALLMARK_NOTCH_SIGNALING'],
             color_pal='YlOrBr')

ggpubr::ggarrange(plotlist=qp, ncol=3, nrow=2, common.legend=TRUE)
```

Although expression of the NOTCH signaling pathway is higher overall in "Lung6",
the expression is not uniform, but rather is arranged in patterns within each FOV.
Conversely, cells in FOVs of "Lung5_Rep1" that show high expression of the NOTCH
signaling pathway are scattered across the tissue.

Since this expression of this pathway has been identified as spatially aggregated,
it might be of interest to know if there is correspondence of NOTCH signaling with
tissue domains. With the `STclust` function, tissue domains can be identified.

```{r stclust_smi, eval=FALSE}
lc_stlist <- STclust(lc_stlist, ws=0.02, ks='dtc')
```

Optionally, if tissue images are available for some or all the FOVs, users can
upload those to the STlist for display next to the other plot types available in
`spatialGE`. Users can upload an entire folder of images to the STlist, as long as
image files have the same names as the FOVs within the STlist (e.g., "Lung5_Rep1_fov_2",
"Lung6_fov_4", etc). Keep in mind, however, this will increase the size of the STlist
significantly, especially in CosMx experiments that usually contain many FOVs.

The multiplexed immunofluorescence images have been previously downloaded from the [spatialGE_Data GitHub repository](https://github.com/FridleyLab/spatialGE_Data). The temporary folder
created at the beginning of this tutorial will be used to generate the file paths
to the folder containing the images. Importantly, the images need to be renamed
according to the FOV name in the STlist. This renaming was already performed for
the files in the spatialGE_Data GitHub repository. Here is an example of how the
files were renamed from the original data set:

- For tissue slice "Lung5_Rep1", images were renamed from "CellComposite_F002.jpg",
"CellComposite_F006.jpg", and "CellComposite_F011.jpg" to "Lung5_Rep1_fov_2.jpg",
"Lung5_Rep1_fov_6.jpg", and "Lung5_Rep1_fov_11.jpg" respectively.
- For tissue slice "Lung6", images were renamed from "CellComposite_F004.jpg",
"CellComposite_F006.jpg", and "CellComposite_F007.jpg" to "Lung6_fov_4.jpg",
"Lung6_fov_6.jpg", and "Lung6_fov_7.jpg" respectively.

Now, upload the images to the STlist with the `load_images` function:

```{r loadimages1, eval = FALSE}
img_fp = '~/Desktop/nanostring_nsclc/Lung5_Rep1-Flat_files_and_images/CellComposite'
lc_stlist <- load_images(lc_stlist, images=img_fp)
```

The images for slide "Lung6" were not found, which are in the corresponding folder. The `load_images` function is ran again to add images from "Lung6":

```{r loadimages2, eval = FALSE}
img_fp = '~/Desktop/nanostring_nsclc/Lung6-Flat_files_and_images/CellComposite'
lc_stlist <- load_images(lc_stlist, images=img_fp)
```

The function plot_image is used to generate the ggplot objects containing the images:
```{r plot_tissues, fig.width=12, eval=FALSE}
ti <- plot_image(lc_stlist)
```

The domains can be plotted with the `STplot` function too.

```{r plot_domains, fig.width=12, eval=FALSE}
dom_p <- STplot(lc_stlist, ks='dtc', ws=0.02, deepSplit=FALSE,
                color_pal='discreterainbow')

ggpubr::ggarrange(dom_p$Lung5_Rep1_fov_2_stclust_spw0.02_dsplFalse,
                  ti$image_Lung5_Rep1_fov_2,
                  dom_p$Lung5_Rep1_fov_6_stclust_spw0.02_dsplFalse,
                  ti$image_Lung5_Rep1_fov_6,
                  dom_p$Lung5_Rep1_fov_11_stclust_spw0.02_dsplFalse,
                  ti$image_Lung5_Rep1_fov_11,
                  dom_p$Lung6_fov_4_stclust_spw0.02_dsplFalse,
                  ti$image_Lung6_fov_4,
                  dom_p$Lung6_fov_6_stclust_spw0.02_dsplFalse,
                  ti$image_Lung6_fov_6,
                  dom_p$Lung6_fov_7_stclust_spw0.02_dsplFalse,
                  ti$image_Lung6_fov_7,
                  ncol=4, nrow=3)

# If no tissue images are available:
# ggpubr::ggarrange(plotlist=dom_p, ncol=3, nrow=2)
```

Now, to identify the domains, differential gene expression can be performed:

```{r domain_de, eval=FALSE}
deg <- STdiff(lc_stlist, w=0.02, k='dtc', deepSplit=FALSE, test_type='wilcoxon')
```

Then, the top 3 DE genes per cluster in the samples showing NOTCH signaling spatial patterns are identified:

```{r top_de, eval=FALSE}
bind_rows(deg) %>%
  filter(adj_p_val < 0.05 & abs(avg_log2fc > 0.05)) %>%
  filter(sample %in% c('Lung6_fov_4', 'Lung6_fov_6', 'Lung6_fov_7')) %>%
  group_by(sample, cluster_1) %>%
  slice_head(n=3)
```

Expression of KRT genes in domains 1 and 4 for FOV "Lung6_fov_4", 1, 4, and 5 for
FOV Lung6_fov_6, and 1, 2, and 5 for FOV is indicative of tumor cells in those domains.
With the exception of 4 in "Lung6_fov_6", there seems to be concordance among the
areas of of NOTCH signaling and the "tumor" domains, which indicates that in slide
"Lung6", the tumor compartment is spatially enriched for NOTCH signaling.

## Identification of genes presenting expression spatial gradients

Since the tumor areas have been visually identified, it is possible now to ascertain
which genes show expression that varies with distance within the tumor compartment.
In the FOVs "Lung6_fov_4", "Lung6_fov_7", and "Lung6_fov_6", the domain 3 seems to
be surrounding the tumor compartment. Hence, domain 3 will be used as reference
domain to test for genes for which expression increases or decreases with distance
from domain 3. This test is performed by the `STgradient` function.

In `STgradient`, the spatial coordinates are used to calculate Euclidean distances of
each cell to each of the other cells. The reference domain is the domain from which
the distances will be calculated (domain 3 in this case). The non-reference domain(s)
are the domain(s) on which the user wants to study gradients. Next, removes isolated
cells in the reference domain (i.e., have a number of immediate neighbors smaller
than parameter `min_nb`). The `min_nb` parameter intends to reduce the effects of
small “pockets” of the reference domain on the correlation coefficient. At this point,
either the minimum or average distance of each non-reference cell to each reference
cell is calculated (`distsumm`). The minimum distance is easier to interpret, however,
it is more sensitive to highly fragmented reference domains. On the other hand, the
interpretability of the average distance is not as easy as the minimum distance,
but it might detect in a better way the gradients when the reference domain is
distributed across the sample. Next, the gene expression data is subset to the
non-reference cells, and the top variable genes (`topgenes`) are selected.
Results will be produced only for those genes. Depending on the user’s selection,
the ordinary least squares (OLS) or robust regression is used to calculate the slope of
the regression line between the minimum/average distance and the expression of a gene.
If OLS was selected, users might opt to remove outliers via the interquartile method.
Finally, the Spearman’s coefficient is calculated in addition to the regression line
coefficient. The slope indicates the direction of the correlation. If positive, the gene
expression tends to be higher when farther from the reference domain. If negative,
gene expression is higher when closer to the reference domain. The interpretation of
the sign for the Spearman’s coefficient is similar, however its magnitude indicates
how strong the correlation is between gene expression and distance to the reference
domain.

The `STgradient` function can be executed with these commands, in order to test
for gradients within the tumor compartment. First, it is necessary to find the name of
the column in `lung_subset@spatial_meta` containing the domain assignments:

```{r columnnames, eval=FALSE}
spatial_metadata(lc_stlist)
```

The name of the column is `stclust_spw0.02_dsplFalse` (STclust results using spatial
weight of 0.02 and DynamicTreeCuts with no deepSplit). This name is used in the
`annot` argument of `STgradient`:

```{r stgradient_run, eval=FALSE}
stg <- STgradient(lc_stlist,
                  topgenes=1000,
                  annot='stclust_spw0.02_dsplFalse',
                  ref=4,
                  samples=c('Lung6_fov_4', 'Lung6_fov_6', 'Lung6_fov_7'),
                  distsumm='min',
                  robust=TRUE)
```

Here are the first few rows of the `STgradient` result for sample "Lung6_fov_4":

```{r stg_res, eval=FALSE}
head(stg[['Lung6_fov_4']])
```

The key columns to look at are `min_spearman_r` and `min_spearman_r_pval_adj`.
Now, to find genes with significant spatial gradients in FOV "Lung6_fov_4", these
commands can be used:

```{r topstg, eval=FALSE}
bind_rows(stg) %>%
  filter(sample_name == 'Lung6_fov_4') %>%
  arrange(desc(abs(min_spearman_r))) %>%
  group_by(sample_name) %>%
  slice_head(n=5) %>%
  ungroup() %>%
  arrange(gene)
```

It can be seen from these results that multiple _KRT6A_ shows spatial gradients.
With a negative Spearman's coefficients, the expression of this gene is higher
in cells close to to domain 4 than those away from it. Conversely, __HLA-B__ shows
higher expression in cells far from domain 4:

```{r stg_plots, eval=FALSE}
qp2 <- STplot(lc_stlist, genes=c('KRT6A', 'HLA-B'), samples='Lung6_fov_4', color_pal='YlOrBr')
ggpubr::ggarrange(plotlist=qp2, common.legend=TRUE)
```

Here are the results for genes within the NOTCH signaling pathway:

```{r stg_notch, eval=FALSE}
bind_rows(stg) %>%
  filter(sample_name == 'Lung6_fov_4') %>%
  arrange(desc(abs(min_spearman_r))) %>%
  filter(gene %in% gene_sets[['HALLMARK_NOTCH_SIGNALING']]) %>%
  filter(min_spearman_r_pval_adj < 0.05)
```

Although the correlations in genes from this pathway are not as strong as and those
for __KRT6A__ and __HLA-B__ (i.e., Spearman's coefficient closer to zero), it can
be seen that the genes __NOTCH1__ and __NOTCH3__ show evidence of expression
gradients in FOV "Lung6_fov_4".
