<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Choosing Genomics Tools: Gene Pattern Activity</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />
<link rel="shortcut icon" href="resources/images/favicon.ico" />
 <!--- go to https://favicon.io/favicon-converter/ to upload an image to make a new favicon.io. You will need to replace the current favicon.io image with the one in the downloaded directory from the website. The current image is in the resources/images/ directory --->

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">University of Michigan Workshops 2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Intro to Reproducibility
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="reproducibility_intro.html">Workshop Material</a>
    </li>
    <li>
      <a href="reproducibility_intro_activity.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Intermediate &amp; Advanced Reproducibility
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="reproducibility_adv.html">Workshop Material</a>
    </li>
    <li>
      <a href="reproducibility_adv_activity.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analyzing Clinical Data
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="clinical_data.html">Workshop Material</a>
    </li>
    <li>
      <a href="clinical_data_activity.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Choosing Genomics Tools
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="choosing_genomics_tools.html">Workshop Material</a>
    </li>
    <li>
      <a href="choosing_genomics_tools_activity.html">Activity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial Transcriptomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="st_tools.html">Workshop Material</a>
    </li>
    <li>
      <a href="st_tools_activity.html">Activity</a>
    </li>
  </ul>
</li>
<li>
  <a href="um_resources.html">UM Resources</a>
</li>
<li>
  <a href="https://www.itcrtraining.org/home">ITN Resources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><strong>Choosing Genomics Tools: Gene
Pattern Activity</strong></h1>

</div>


<div id="bulk-rna-seq-differential-expression-analysis"
class="section level2">
<h2>Bulk RNA-seq differential expression analysis</h2>
<p>The datasets for this activity consist of 20 Breast cancer primary
tumor samples with 20 normal matched samples from TCGA as explained in
the study <a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC9028992/">“Molecular
analysis of TCGA breast cancer histologic types”</a>.</p>
<p>The aim of this activity is to answer the question – “What genes
differentiate samples with breast cancer from samples without breast
cancer?”. This is an example of differential expression analysis, a
downstream analysis step in the bulk RNA-seq workflow.</p>
<div id="download-data-and-log-into-the-genepattern-platform"
class="section level3">
<h3>Download Data and log into the GenePattern platform</h3>
<p><input type="checkbox"> <a
href="https://datasets.genepattern.org/?prefix=data/workshops/240411-PSTP/">Download
the relevant datasets.</a><br/> - <input type="checkbox"> Download
<code>BRCA_HUGO_symbols.preprocessed.cls</code><br/> -
<input type="checkbox"> Download
<code>BRCA_HUGO_symbols.preprocessed.gct</code><br/></p>
<p><em>Note, clicking on the filenames should automatically download
them, otherwise you might need to right click.</em></p>
<p><a href="https://datasets.genepattern.org/?prefix=data/workshops/240411-PSTP/" target="_blank"><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_0.png" alt="Download the c l s and g c t files from the gene pattern datasets resource, specifically the 2 4 0 4 1 1 hyphen P S T P workshops directory" width="480" /></a></p>
<p><em>Make sure you know where they are saved on your computer as
you’ll be uploading them in an upcoming step.</em></p>
<details>
<summary>
What are these file types?
</summary>
<p>The <code>gct</code> file is a Gene Cluster Text file format used to
describe an expression dataset. <a
href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GCT:_Gene_Cluster_Text_file_format_.28.2A.gct.29">See
more about <code>gct</code> files at this Broad Institute
resource.</a></p>
<p><a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GCT:_Gene_Cluster_Text_file_format_.28.2A.gct.29" target="_blank"><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_11.png" alt="Description of g c t file from the Broad Institute" width="480" /></a></p>
<p>The <code>cls</code> file is a Categorial (e.g., tumor vs normal)
class file format, specifying phenotype or class labels associated with
each sample in the expression data. <a
href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29">See
more about <code>cls</code> files at this Broad Institute
resource.</a></p>
<p><a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29" target="_blank"><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_19.png" alt="Description of c l s file from the Broad Institute" width="480" /></a></p>
<p><a
href="https://www.genepattern.org/file-formats-guide/#gsc.tab=0">See
this resource from GenePattern for more information on creating these
input files for your work.</a></p>
<p><a href="https://www.genepattern.org/file-formats-guide/#gsc.tab=0" target="_blank"><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_5.png" alt="Description of c l s and g c t file types with a link to a resource on creating these file types for your work" width="480" /></a></p>
</details>
<p><input type="checkbox"> <a href="http://cloud.genepattern.org/gp/">Go
to the Gene Pattern platform</a> and sign in.</p>
</div>
<div
id="run-the-comparativemarkerselection-module-to-identify-differentially-expressed-genes-between-the-two-groups"
class="section level3">
<h3>Run the ComparativeMarkerSelection Module to identify differentially
expressed genes between the two groups</h3>
<p><input type="checkbox"> Find the “ComparativeMarkerSelection” Module
by searching for it in the search tool.</p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_26.png" alt="Find the Comparative Marker Selection Module to identify differentially expressed genes that can discriminate between our two groups of data" width="480" /></p>
<p><input type="checkbox"> Select the <code>gct</code> file
(<code>BRCA_HUGO_symbols.preprocessed.gct</code>) for the “Input file*”
option.</p>
<p><input type="checkbox"> Select the <code>cls</code> file
(<code>BRCA_HUGO_symbols.preprocessed.cls</code>) for the “cls file*”
option.</p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_41.png" alt="select the input files" width="480" /></p>
<p><input type="checkbox"> Scroll down and click the Run button.</p>
<p><input type="checkbox"> Click on the produced <code>odf</code> file
(<code>BRCA_HUGO_symbols_preprocessed.comp.marker.odf</code>).</p>
</div>
<div
id="run-the-differentialexpressionviewer-module-to-make-a-heatmap-of-the-differentially-expressed-genes"
class="section level3">
<h3>Run the DifferentialExpressionViewer module to make a heatmap of the
differentially expressed genes</h3>
<p><input type="checkbox"> Click on the produced file from the
comparative marker selection.
<img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_50.png" alt="Click on the produced file from the comparative marker selection" width="480" /></p>
<details>
<summary>
What is this file type?
</summary>
<p>An ODF is an Output Description Format file, a GenePattern specific
filetype. It is similar to a GCT file format, describing a gene
expression dataset; however, the ODF filetype has a specialized header.
<a href="https://www.genepattern.org/file-formats-guide/#ODF">See more
information about ODF files from GenePattern.</a></p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2800f2a6905_0_122.png" alt="The O D F file is a Gene Pattern specific file type that is similar to a G C T file type, but has a specialized header" width="480" /></p>
</details>
<p><input type="checkbox"> In the opened side panel, scroll down and
under the “Send to Module” heading, click on
“DifferentialExpressionViewer”.</p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_55.png" alt="Send the data to the module named differential expression viewer" width="480" /></p>
<p><input type="checkbox"> Confirm that the ODF file has already been
selected for the “differential expression filename<em>”.
<input type="checkbox"> Click the upload file button for the ”dataset
filename</em>” section. <input type="checkbox"> Select the
<code>gct</code> file
(<code>BRCA_HUGO_symbols.preprocessed.gct</code>).</p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_66.png" alt="select the input files" width="480" /></p>
<p><input type="checkbox"> Click the Run button.</p>
</div>
<div id="inspect-and-customize-the-heatmap" class="section level3">
<h3>Inspect and customize the heatmap</h3>
<p>Inspecting the heatmap we see gene names are the row names and sample
names are the column names.</p>
<p>Color-wise there are blues and reds and whites. But without a legend,
what do these colors mean?</p>
<p><input type="checkbox"> Click on the “Edit” dropdown menu in the top
left.</p>
<p><input type="checkbox"> Click on “Display Options”.</p>
<p><input type="checkbox"> Click the checkbox next to “Display
Legend”</p>
<p><input type="checkbox"> Click “OK”.</p>
<p>From this newly added legend, we see that red is a “row max” value
while blue is a “row min” value.</p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2800f2a6905_0_160.png" alt="legend values" width="480" /></p>
<p>If a gene has very red colors across normal tissue samples and very
blue colors across primary tissues – that translates to that gene being
upregulated in the matched normal tissues. MME is an example of this.
(MME could also be termed downregulated in primary tumor samples).</p>
<p>To save the heatmap (specifically those genes identified as
comparative markers or differentially expressed genes) ….</p>
<p><input type="checkbox"> Click on the “File” dropdown menu in the top
left.</p>
<p><input type="checkbox"> Click on “Save Image”.</p>
<p><input type="checkbox"> You can select the “File type:” you want.
<code>svg</code> is default. <code>png</code> is the other option.</p>
<p><img src="choosing_genomics_tools_activity_files/figure-html/1VBriRWSN-XJEjuzP29TkbkpiGpSJslTrtleqZwtlsDM_g2f2ca03ec26_0_79.png" alt="saving the heatmap image" width="480" /></p>
<p>If you click the “Include” all features and samples checkbox, it will
include all genes whether they were identified as differentially
expressed or not, producing a much larger heatmap.</p>
<p><input type="checkbox"> Click on “OK”.</p>
<p>Fold Change and statistical test statistics (e.g., significance) are
provided in a related table that you can also download.</p>
<p><input type="checkbox"> Click on the “File” dropdown menu in the top
left.</p>
<p><input type="checkbox"> Click on the “Save Table (.txt)”</p>
<p><input type="checkbox"> If you want to save it to your computer,
click on “Download” and select “OK”.</p>
<!--
## Single cell RNA-seq

- [Download practice 10x single cell data](https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz)
- [Go to the Gene Pattern platform](http://cloud.genepattern.org/gp/)
-->
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
